In [10]:
import pickle
import pandas as pd
import joblib

In [7]:
input_data = {
    'Age': 35,
    'DailyRate': 800,
    'DistanceFromHome': 5,
    'Education': 3,
    'EnvironmentSatisfaction': 4,
    'HourlyRate': 60,
    'JobInvolvement': 3,
    'JobLevel': 2,
    'JobSatisfaction': 4,
    'MonthlyIncome': 6000,
    'MonthlyRate': 12000,
    'NumCompaniesWorked': 1,
    'PercentSalaryHike': 14,
    'PerformanceRating': 3,
    'RelationshipSatisfaction': 3,
    'StockOptionLevel': 1,
    'TotalWorkingYears': 10,
    'TrainingTimesLastYear': 3,
    'WorkLifeBalance': 2,
    'YearsAtCompany': 5,
    'YearsInCurrentRole': 3,
    'YearsSinceLastPromotion': 1,
    'YearsWithCurrManager': 2,
    # One-hot encoded fields (sample values)
    'BusinessTravel_Non-Travel': False,
    'BusinessTravel_Travel_Frequently': True,
    'BusinessTravel_Travel_Rarely': False,
    'Department_Human Resources': False,
    'Department_Research & Development': True,
    'Department_Sales': False,
    'EducationField_Human Resources': False,
    'EducationField_Life Sciences': True,
    'EducationField_Marketing': False,
    'EducationField_Medical': False,
    'EducationField_Other': False,
    'EducationField_Technical Degree': False,
    'Gender_Female': False,
    'Gender_Male': True,
    'JobRole_Healthcare Representative': False,
    'JobRole_Human Resources': False,
    'JobRole_Laboratory Technician': False,
    'JobRole_Manager': True,
    'JobRole_Manufacturing Director': False,
    'JobRole_Research Director': False,
    'JobRole_Research Scientist': False,
    'JobRole_Sales Executive': False,
    'JobRole_Sales Representative': False,
    'MaritalStatus_Divorced': False,
    'MaritalStatus_Married': True,
    'MaritalStatus_Single': False,
    'OverTime_No': False,
    'OverTime_Yes': True,
}


In [8]:
# Convert to DataFrame
input_df = pd.DataFrame([input_data])

In [12]:
# Load Model
model_filename = 'Random Forest_model.pkl'
model = joblib.load(model_filename)

In [13]:
# Predict and Output
prediction = model.predict(input_df)[0]
probability = model.predict_proba(input_df)[0]

print(f"Prediksi: {prediction}")
print("Probabilitas per kelas:")
for i, prob in enumerate(probability):
    print(f"  Kelas {i}: {prob:.4f}")

Prediksi: 0
Probabilitas per kelas:
  Kelas 0: 0.8200
  Kelas 1: 0.1800


Kelas 0 artinya karyawan yang tidak mengundurkan diri, sedangkan kelas 1 adalah karyawan yang mengundurkan diri.

Berdasarkan output, prediksi dari data di atas adalah 0 atau karyawan tidak mengundurkan diri dengan probabilitas 0.82.